src/d3a/models/market/one_sided.py

```diff
@@ -90,23 +94,34 @@ def delete_offer(self, offer_or_id: Union[str, Offer]):
         # TODO: Once we add event-driven blockchain, this should be asynchronous
         self._notify_listeners(MarketEvent.OFFER_DELETED, offer=offer)
 
+    def _update_fee_and_calculate_final_price(self, energy, trade_rate,
+                                              energy_portion, original_price):
+        fees = self.transfer_fee_ratio * original_price * energy_portion
+        fees += self.transfer_fee_const * energy
+        self._grid_fee += fees
+        return energy * trade_rate + fees
```

Comment:
I like the level of verbosity of this function for understanding the code. However, it will be called quite a lot during a simulation, right? Wouldn't it be better ti reduce the number of calculations (lines) to reduce computational effort? Or doesn't it make a big difference?

Reply:
Hmm, I really do not know how the Python interpreter optimises arithmetic operations. If we were talking about compilers, gcc or clang automatically optimise and merge these 3 operations to one (by producing assembly code that sends these operations at the CPU on the same cycle) but I do not really know how the interpreter deals with this. Just to be on the safe side, I will follow your suggestion and calculate the fees variable on one line. Thanks!